src/lib/results.ts

```diff
@@ -1,32 +1,106 @@
 import { NotifyResult } from './ic';
-import { Alias, Variant, nat64, nat } from './candid_types';
+import { Alias, Variant, nat64, nat, Service } from './candid_types';
 import { RequireExactlyOne } from './candid_types/variant';
 
+/**
+ * Represents either success (`Ok`) or failure (`Err`)
+ *
+ * @typeParam Ok - The type of the success value
+ * @typeParam Err - The type of the error value
+ */
 export type Result<Ok, Err> = Variant<{
+    /** The success value */
     Ok: Ok;
+    /** The error value */
     Err: Err;
 }>;
 
 export const Result = {
+    /**
+     * Wraps the provided value in an `Ok` {@link Result}
+     * @param value - the success value to be wrapped
+     * @returns a successful {@link Result} containing the provided value
+     */
     Ok: <Ok, Err>(value: Ok): Result<Ok, Err> => ({
         Ok: value
     }),
+    /**
+     * Wraps the provided value in an `Err` {@link Result}
+     * @param value - the error value to be wrapped
+     * @returns an error {@link Result} containing the provided value
+     */
     Err: <Ok, Err>(value: Err): Result<Ok, Err> => ({
         Err: value
     })
 };
 
+/**
+ * An execution context for configuring {@link Service} method calls.
+ *
+ * Each service method call can await a result ({@link CallResult.call}) or can
+ * be sent without awaiting a response ({@link CallResult.notify}).
+ *
+ * To send cycles with your request first chain either {@link CallResult.cycles}
+ * or {@link CallResult.cycles128}.
+ */
 export type CallResult<T> = {
+    /**
+     * Dispatches the asynchronous {@link Service} method call and awaits a
```

Comment:
It doesn't await a response (depending on how you think about it, I think it might be confusing to say this), you must await the response. I think we should just mention that this returns a promise that can be awaited